local/src/core/ProxyVerifier.cc

```diff
@@ -1663,22 +1823,46 @@ std::string HttpHeader::make_key() const {
   return std::move(key);
 }
 
-bool HttpHeader::verify_headers(swoc::TextView key,
+// Verify that the fields in 'this' correspond to the provided rules.
+bool HttpHeader::verify_headers(swoc::TextView transaction_key,
                                 const HttpFields &rules_) const {
   // Remains false if no issue is observed
   // Setting true does not break loop because test() calls errata.diag()
   bool issue_exists = false;
-  for (const auto &rule : rules_._rules) {
-    // Hashing uses strcasecmp internally
-    auto found_iter = _fields_rules->_fields.find(rule.first);
-    if (found_iter == _fields_rules->_fields.cend()) {
-      if (!rule.second->test(key, swoc::TextView(), swoc::TextView())) {
-        issue_exists = true;
+  auto const& rules = rules_._rules;
+  auto const& fields = _fields_rules->_fields;
+  for (auto const &[name, rule_check] : rules) {
+    auto name_range = fields.equal_range(name);
+    auto field_iter = name_range.first;
+    if (rule_check->expects_duplicate_fields()) {
+      std::list<TextView> values;
```

Comment:
Why is this declared and passed in? Why isn't it a local variable in `test`?

Reply:
It's initialized here as empty and passed in empty, intentionally, in the case where there are no fields. But in the case where there are fields it is populated.

As discussed offline, I'll instead explicitly pass in an empty list when the list is empty and push this declaration down to the place where there are fields and it is populated.